dulwich/tests/test_porcelain.py

```diff
@@ -702,6 +706,81 @@ def test_empty(self):
                             outstream=outstream)
         self.assertEqual(outstream.getvalue(), b"")
 
+    def test_diff_apply(self):
```

Comment:
This test is verifying compatibility with C git; please move it somewhere under dulwich/tests/compat. Tests not under compat/ should not require git itself to be installed.